lib/bridge/IrcHandler.js

```diff
@@ -9,13 +9,22 @@ var MatrixRoom = require("matrix-appservice-bridge").MatrixRoom;
 var MatrixUser = require("matrix-appservice-bridge").MatrixUser;
 var MatrixAction = require("../models/MatrixAction");
 
+var Queue = require("../util/Queue.js");
+
 function IrcHandler(ircBridge) {
     this.ircBridge = ircBridge;
     // maintain a map of which user ID is in which PM room, so we know if we
     // need to re-invite them if they bail.
     this._roomIdToPrivateMember = {
         // room_id: { user_id: $USER_ID, membership: "join|invite|leave|etc" }
     };
+
+    // Use a queue to keep the setting of topics in rooms atomic in order to
+    // prevent races involving several topics being received from IRC in quick
+    // succession. This queue will cause them to block unless the server, channel
+    // and topic are the same as an ongoing promise to set a topic, whihc will in
+    // that case be returned.
```

Comment:
`whihc`

Also, last sentence is a bit hard to grok, you're describe a 3-tuple with words - I would just say:

> This queue will cause them to block unless (server, channel, topic) are the same [...]